core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp

```diff
@@ -1063,6 +1171,37 @@ imm_set_value(Imm &imm, void *data, uint32 bytes)
     }
 }
 
+#if WASM_ENABLE_SHARED_MEMORY != 0
+static uint32
+mov_imm_to_free_reg(x86::Assembler &a, Imm &imm, uint32 bytes)
+{
+    uint32 reg_no;
+
+    switch (bytes) {
+        case 1:
+            reg_no = REG_I8_FREE_IDX;
+            a.mov(regs_i8[reg_no], imm);
```

Comment:
Share we consider the higher 24bit of regs_i8[reg_no]?

Reply:
It should be fine since we only move the value in the same register into memory. For example, we move 1-byte imm into `regs_i8[REG_I8_FREE_IDX]`, later in `xchg_r_to_m()`, we will call `a.xchg(m_dst, regs_i8[reg_no_src]);` to store the 1-byte value inside the register `regs_i8[REG_I8_FREE_IDX]` to memory